Core/NoOpRunner.Core/RandomNumber.cs

```diff
@@ -0,0 +1,40 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace NoOpRunner.Core
+{
+    class RandomNumber
```

Comment:
If we do need this singleton, use the [Lazy singleton pattern](https://riptutorial.com/csharp/example/6795/lazy--thread-safe-singleton--using-lazy-t--), no need for manual locks (unless we need the number generation itself to be thread-safe, then you'd also need to sync GetRandom)